api/util/apps.go

```diff
@@ -229,3 +241,51 @@ func GatherDNSDetails(items []apps.Application) []DNSDetail {
 	}
 	return result
 }
+
+// ValidatePEM validates if the passed content is in valid PEM format, errors
+// out if the content is empty
+func ValidatePEM(content string) (*string, error) {
+	if content == "" {
+		return nil, fmt.Errorf("the SSH private key cannot be empty")
+	}
+
+	content = strings.TrimSpace(content)
+	b, rest := pem.Decode([]byte(content))
+	if b == nil || len(rest) > 0 {
+		return nil, fmt.Errorf("no valid PEM formatted data found")
+	}
+	return &content, nil
+}
+
+// TestRepositoryAccess tests if the given git repository can be accessed.
+func TestRepositoryAccess(ctx context.Context, client *GitInformationClient, git *apps.ApplicationGitConfig, auth GitAuth) error {
+	repoInfo, err := client.RepositoryInformation(ctx, git.URL, auth)
+	if err != nil {
+		// we are not returning a detailed error here as it might be
+		// too technical. The full error can still be seen by using
+		// a different RetryLog function in the client.
+		return errors.New(
+			"communication issue with git information service " +
+				"(use --skip-repo-access-check to skip this check)",
+		)
+	}
+	if len(repoInfo.Warnings) > 0 {
+		fmt.Fprintf(os.Stderr, "warning: %s\n", strings.Join(repoInfo.Warnings, "."))
+	}
+	if repoInfo.Error != "" {
+		return errors.New(repoInfo.Error)
+	}
+	if !(containsBranch(git.Revision, repoInfo) ||
+		containsTag(git.Revision, repoInfo)) {
+		return fmt.Errorf(
+			"can not find specified git revision (%q) in repository",
+			git.Revision,
```

Comment:
Would be kind of nice if this error would display the revisions it found in the repo. But I guess it could get quite long if there are lots of branches/tags :smile:

Reply:
Yeah was thinking about this too, but then in the end the important message is that the revision you specified is not in. It would be nice if something like "did you mean <suggested revision>" would be shown, but this is a bit out of scope I'd say.

Reply:
yeah, fuzzy matching and then `did you mean x` would probably the best user experience. Agreed that it is out of scope for now.